### ISSUE TITLE
day1

### ISSUE THREAD

**Issue by sanshengshi626** [sanshengshi626/daka-basic-machine-learning#1]
1. 机器学习的一些概念 有监督、无监督、泛化能力、过拟合欠拟合(方差和偏差以及各自解决办法)、交叉验证
有监督：有预测标签
无监督：无标记 找规律
泛化能力：对新样本的适应能力
过拟合：training很好 testing不好
欠拟合：模型没有很好地捕捉到数据特征，不能够很好地拟合数据
交叉验证：把数据分n分 拿n-1训练 1份测试 重复n次 求最后平均值
方差和偏差以及各自解决办法：variance特别大 模型过拟合 波动较大 bias较大 模型欠拟合 和预测值相差较远 
解决：偏差：增加feature 寻找有用的feature 增加model复杂性 方差：减少feature 减少model复杂性 增加data
2. 线性回归的原理
y=b0+b1x1+b2x2+......
3. 线性回归损失函数、代价函数、目标函数
损失函数：MSE/RMSE sum(yi-y'i)^2/n
4. 优化方法(梯度下降法、牛顿法、拟牛顿法等)
梯度下降法：找一个点，求相应梯度，用一个小步长，反方向下降，重复直到到最低点
牛顿法：选择一个接近函数 {\displaystyle f(x)} f(x)零点的 {\displaystyle x_{0}} x_{0}，计算相应的 {\displaystyle f(x_{0})} f(x_0)和切线斜率 {\displaystyle f'(x_{0})} f'(x_0)（这里 {\displaystyle f'} f'表示函数 {\displaystyle f} f的导数）。然后我们计算穿过点 {\displaystyle (x_{0},f(x_{0}))} (x_{0},f(x_{0}))并且斜率为 {\displaystyle f'(x_{0})} f'(x_0)的直线和 {\displaystyle x} x轴的交点的 {\displaystyle x} x坐标
我们将新求得的点的 {\displaystyle x} x坐标命名为 {\displaystyle x_{1}} x_1，通常 {\displaystyle x_{1}} x_1会比 {\displaystyle x_{0}} x_{0}更接近方程 {\displaystyle f(x)=0} f(x)=0的解。因此我们现在可以利用 {\displaystyle x_{1}} x_1开始下一轮迭代
拟牛顿法：
在牛顿法的迭代中，需要计算海森矩阵的逆矩阵 [公式] ，这一计算比较复杂，考虑用一个 [公式] 阶矩阵 [公式] 来近似代替 [公式]
5. 线性回归的评估指标
MAE=1n∑i=1n|fi−yi|
MSE=1n∑i=1n(fi−yi)2
RMSE=MSE‾‾‾‾‾√
MAPE=100n∑t=1n|yi−fiyi
r2=1−SSresSStot=1−∑(yi−fi)2∑(yi−y¯)2
adjust r2
6. sklearn参数详解
sklearn.linear_model.LinearRegression
fit_intercept ：whether to calculate the intercept for this mode
normalize:the regressors X will be normalized before regression by subtracting the mean and dividing by the l2-norm.
